README.md

```diff
@@ -0,0 +1,209 @@
+* `MaxMind::GeoIP2::Record::City` - `city.geoname_id`
+* `MaxMind::GeoIP2::Record::Continent` - `continent.code` or
+  `continent.geoname_id`
+* `MaxMind::GeoIP2::Record::Country` and
+  `MaxMind::GeoIP2::Record::RepresentedCountry` - `country.iso_code` or
+  `country.geoname_id`
+* `MaxMind::GeoIP2::Record::Subdivision` - `subdivision.iso_code` or
+  `subdivision.geoname_id`
+
+### What data is returned?
+
+While many of the end points return the same basic records, the attributes
+which can be populated vary between end points. In addition, while an end
+point may offer a particular piece of data, MaxMind does not always have
+every piece of data for any given IP address.
+
+Because of these factors, it is possible for any end point to return a
+record where some or all of the attributes are unpopulated.
+
+See the [GeoIP2 Precision web service
+docs](https://dev.maxmind.com/geoip/geoip2/web-services) for details on
+what data each end point may return.
+
+The only piece of data which is always returned is the `ip_address`
+attribute in the `MaxMind::GeoIP2::Record::Traits` record.
+
+## Integration with GeoNames
+
+[GeoNames](https://www.geonames.org/) offers web services and downloadable
+databases with data on geographical features around the world, including
+populated places. They offer both free and paid premium data. Each feature
+is unique identified by a `geoname_id`, which is an integer.
+
+Many of the records returned by the GeoIP2 web services and databases
+include a `geoname_id` property. This is the ID of a geographical feature
+(city, region, country, etc.) in the GeoNames database.
+
+Some of the data that MaxMind provides is also sourced from GeoNames. We
+source things like place names, ISO codes, and other similar data from the
+GeoNames premium data set.
+
+## Reporting data problems
+
+If the problem you find is that an IP address is incorrectly mapped, please
+[submit your correction to MaxMind](https://www.maxmind.com/en/correction).
+
+If you find some other sort of mistake, like an incorrect spelling, please
+check the [GeoNames site](https://www.geonames.org/) first. Once you've
+searched for a place and found it on the GeoNames map view, there are a
+number of links you can use to correct data ("move", "edit", "alternate
+names", etc.). Once the correction is part of the GeoNames data set, it
+will be automatically incorporated into future MaxMind releases.
+
+If you are a paying MaxMind customer and you're not sure where to submit a
+correction, please [contact MaxMind
+support](https://www.maxmind.com/en/support) for help.
+
+## Support
+
+Please report all issues with this code using the [GitHub issue
+tracker](https://github.com/maxmind/GeoIP2-ruby/issues).
+
+If you are having an issue with a MaxMind service that is not specific to the
+client API, please see [our support page](https://www.maxmind.com/en/support).
+
+## Requirements
+
+This code requires Ruby version 2.4 or higher. Older versions may work, but
```

Comment:
Will they work? Doesn't the Gem require 2.4? I suppose if you install it manually, it could work.

Reply:
Yeah I don't think it would work actually. I'll drop that!